GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+handler.
+
+## Promise
+
+The 30-second version:
+
+A [well-specified][promises-aplus] pattern that provides a container type
+designed to describe the dependence graph of asynchronous operations within a
+program. As a container type, any value `T` may be cast to an asynchronous
+value `Promise<T>`. The process of casting `T` to `Promise<T>` is known as
+"resolution". Any value `Promise<T>` may be unwrapped to `T` by passing a
+handler (defined below) to `.then`. Unwrapping via `.then` creates a new
+`Promise` that will resolve to the return value of the handler. One promise
+may have zero-to-many dependent promises.
+
+Errors will propagate from a source promise to all child promises. This process
+is known as "rejection". If a rejected promise has no children, it is
+considered an unhandled rejection. It may be handled in a subsequent operation.
+If a promise has an error handler, propagation from source promises will halt
+at that promise. If a promise has many children, and only one handles errors,
+the other children will still be reported as unhandled rejections.
+
+Unwrapping `Promise<Promise<T>>` will automatically unwrap to `T`. That is:
+`.then(() => Promise.resolve(3)).then(val => console.log(val))` will
+output `3`.
+
+Internally, a promise is represented by a state machine that may be in one of
+three states — pending, resolved, or rejected. The resolved and rejected states
+are collectively referred to as "settled." Once a promise is settled, it cannot
+transition back into pending state.
+
+### Executor / Resolver
+
+The function passed to the `new Promise` constructor. Run **synchronously**.
+Not in common direct use by most promise-users — used mainly to wrap sources of
+asynchrony. As such, it rarely resolves or rejects on the same tick.
+
+Catches all errors thrown. If an error is caught, the returned promise will be
+**rejected**. Because it starts life as a rejected promise with no children, it
+will immediately hit the `OnUnhandledRejection` callback.
+
+```js
+new Promise(function Excecutor (resolve, reject) {
+  // run immediately.
+})
+```
+
+### Handler
+
+A function passed to `Promise#then` or `Promise#catch`. Receives as an argument
+the parent promise's resolved value or rejected error. Represents a new
+`Promise`. A handler which returns a value `T` or `Promise<T>` will resolve the
+new promise with `T`. A handler which throws an error `R`, or returns a
+`Promise` which itself eventually rejects (or has already rejected) with `R`,
+will reject the new promise with `R`.
+
+```js
+
+const p = new Promise(() => {})
+
+const p2 = p.then(function SuccessHandler(value) {
+  // fired if p resolves,
+}, function ErrorHandler(err) {
+  // fired if p rejects.
+})
+
+const p3 = p2.catch(function ErrorHandler(value) {
+  // fired if p2 rejects.
+})
+
+const p4 = p3.then(function SuccessHandler (value) { 
+  // resolved with p2's value if p2 resolves,
+  // resolved with p3's value if p2 rejects and p3 resolves
+})
+```
```

Comment:
We can do that, but I'm specifically talking about synchronous try/catch here. Can be solved at a later pr though.